sections/findify-recommendation.liquid

```diff
@@ -34,42 +34,74 @@
 
       
 <script id="findify-recommendation-{{recommendation_id}}-done" >
-
-  /*
+    /*
   * This script block will be executed at the end of every Findify render cycle.
   * This is the ideal place where to bind all the events of the DOM, such as filters or button events.
   */
+  const selector = "#{{ recommendation_id  }}";
+  const meta =  Findify?.state?.recommendations['{{ recommendation_id }}']?.data.meta;
+  if( !meta ) throw new Error('There is no meta for recommendation widget id = {{ recommendation_id }}')      
+  const rid = meta.rid;
 
-    const swiper = new Swiper("#{{ recommendation_id  }}", {
-      direction: 'horizontal',
-      loop: true,
-      pagination: {
-        el: '.swiper-pagination',
-        type: 'bullets',
-        clickable:true,
-        dynamicBullets: true,
-        dynamicMainBullets: 4
-      },
-      slidesPerView:   "{{ desktop_grid_size }}",
-      breakpoints: {
-        320: {
-          slidesPerView: 2,
-          spaceBetween: 20
-        },
-        480: {
-          slidesPerView: 3,
-          spaceBetween: 30
-        },
-        640: {
-          slidesPerView:  "{{ desktop_grid_size }}",
-          spaceBetween: 30
-        }
+  const swiper = new Swiper(selector, {
+    direction: 'horizontal',
+    loop: true,
+    pagination: {
+      el: '.swiper-pagination',
+      type: 'bullets',
+      clickable:true,
+      dynamicBullets: true,
+      dynamicMainBullets: 4
+    },
+    slidesPerView:   "{{ desktop_grid_size }}",
+    breakpoints: {
+      320: {
+        slidesPerView: 2,
+        spaceBetween: 20
       },
-      navigation: {
-        nextEl: '.swiper-button-next',
-        prevEl: '.swiper-button-prev',
+      480: {
+        slidesPerView: 3,
+        spaceBetween: 30
       },
-    
-    });
+      640: {
+        slidesPerView:  "{{ desktop_grid_size }}",
+        spaceBetween: 30
+      }
+    },
+    navigation: {
+      nextEl: '.swiper-button-next',
+      prevEl: '.swiper-button-prev',
+    },
+
+  });
 
+  const navigate = (url, e) => {
```

Comment:
As we need just the Analytics to be fired.

Wouldn't be good enough to bind the click for analytics and for navigation rely on a <a tag?